detox/src/Detox.js

```diff
@@ -52,7 +52,7 @@ class Detox {
     this._client = new Client(sessionConfig);
     await this._client.connect();
 
-    const DeviceDriverClass = DEVICE_CLASSES[this._deviceConfig.type];
+    const DeviceDriverClass = DEVICE_CLASSES[this._deviceConfig.type] || require(this._deviceConfig.type);
```

Comment:
One thing I don't like about this direct fallback to require is you get a weird error about module resolution if you have a typo for one of the built-in drivers. In other words, the `if` on the line below is sort of useless...

@rotemmiz, what are your thoughts about requiring third party drivers to be prefixed in the configuration?

Something like:

```
"ios.sim.release": {
        "binaryPath": "...",
        "build": "...",
        "type": "plugin:my-custom-driver-package-name",
        "device": {
	  // can be pretty flexible (according to the device driver), add whatever device specification needed
        }
      },
```

```
    let DeviceDriverClass = DEVICE_CLASSES[this._deviceConfig.type];
    if (!DeviceDriverClass) {
      if (this._deviceConfig.type.startsWith('plugin:')) {
            DeviceDriverClass = require(this._deviceConfig.type);
      }
      throw new Error(`'${this._deviceConfig.type}' is not supported`);
    }
```